### PR TITLE
fix(KONFLUX-3991): sign manifest list digests

### PR DESCRIPTION
**Pull request by scoheb**
- needed for bootc images

### CODE REVIEW

**Review by happybhati — CHANGES_REQUESTED** [review]
Task version bump and readme changes are missing.

**Review by davidmogar — COMMENTED** [review]
LGTM but please take care of Happy's problem before this is merged.

File: tasks/rh-sign-image/rh-sign-image.yaml [hunk 1]
```diff
@@ -92,7 +92,12 @@ spec:
             if [ $(jq -r '.mediaType' <<< $RAW_OUTPUT) == "application/vnd.oci.image.manifest.v1+json" ] ; then
               manifest_digests="${referenceContainerImage#*@}"
             else
+              manifest_list_digest="${referenceContainerImage#*@}"
               manifest_digests=$(jq -r '.manifests[].digest' <<< $RAW_OUTPUT)
+              #
+              # add manifest list digest to list of digests to sign
+              #
+              manifest_digests="$manifest_digests $manifest_list_digest"
             fi
```
Comment (mmalina) [inline tasks/rh-sign-image/rh-sign-image.yaml]:
```suggestion
            # Always sign the top level sha
            manifest_digests="${referenceContainerImage#*@}"
            # For multi arch, also sign all the manifests inside
            if [ $(jq -r '.mediaType' <<< $RAW_OUTPUT) != "application/vnd.oci.image.manifest.v1+json" ] ; then
              nested_digests=$(jq -r '.manifests[].digest' <<< $RAW_OUTPUT)
              manifest_digests="$manifest_digests $nested_digests"
            fi
```

Comment (mmalina) [reply tasks/rh-sign-image/rh-sign-image.yaml]: In my proposal the order is different - the manifest list is first. So either you'd have to change the test, or you could also change the assignment. I think it's nice to start with the top level sha, so changing the test is preferable.

**Comment by scoheb** [comment]
> Task version bump and readme changes are missing.

good catch. thanks

File: tasks/rh-sign-image/rh-sign-image.yaml (hunk 1 above)
Comment (scoheb) [reply tasks/rh-sign-image/rh-sign-image.yaml]: I like your proposal!

**Review by mmalina — APPROVED** [review]

File: tasks/rh-sign-image/rh-sign-image.yaml (hunk 1 above)
Comment (mmalina) [reply tasks/rh-sign-image/rh-sign-image.yaml]: Cool!

**Review by happybhati — APPROVED** [review]

**Pull request merged**